packages/app/src/cli/services/context/breakdown-extensions.ts

```diff
@@ -0,0 +1,264 @@
+    existingFieldNames,
+    existingUpdatedFieldNames,
+    newFieldNames,
+    deletedFieldNames,
+  }
+}
+
+/**
+ * This method extracts the list of global fields or global sections from the string that represents a toml section like
+ * this:
+ *        embedded = true
+ *
+ *        [access_scopes]
+ *        scopes = "read_products,write_products,write_discounts"
+ *
+ *        [webhooks.privacy_compliance]
+ *        customer_deletion_url = "https://myhooks.dev/apps/customer_deletion_url_edited"
+ *
+ * Each block is separated by a breaking line. The method will the extract
+ * the `field`  following these patterns:
+ * - <field> = <value> (in this case all the fields inside the block that matches the pattern will be returned)
+ * - [<field>]
+ * - [\<field.subsection\>]
+ *
+ * @param diffConfigContent - The toml string to parse
+ * @returns The list of fields
+ */
+function getFieldsFromDiffConfigContent(diffConfigContent: string): string[] {
+  const fields = diffConfigContent
+    // Split the input string into sections by one or more blank lines
+    .split(/\n\s*\n/)
+    .flatMap((section) => {
+      // Split each section into lines
+      const lines = section.split('\n')
+      if (lines.length === 0) return []
+      // Match the first line of the section against a regular expression to extract the first field name based on the
+      // described patterns
+      const firstLineMatch = lines[0]!.match(/^(?:\[(\w+)|(\w+)\s*=)/)
+      if (!firstLineMatch) return []
+      // Extract the first field name from the appropriate capture group
+      const firstFieldName = firstLineMatch[1] || firstLineMatch[2]
+      if (!firstFieldName) return []
+      // Return field if matches either the pattern [\<field.subsection\>] or [<field>]
+      if (firstFieldName.includes('.')) return [firstFieldName.split('.')[0]]
+      // If the first line of the section matches the pattern  <field> = <value> extract the following
+      // <field> = <value>  that match that condition until the section is finished
+      const otherFieldNames = firstLineMatch[2]
+        ? lines
+            .slice(1)
+            .map((line) => line.match(/^(\w+)\s*=/))
+            .filter(Boolean)
+            .map((match) => match![1])
+        : []
+      return [firstFieldName, ...otherFieldNames]
+    })
+    .filter((match): match is string => match !== undefined)
+
+  // Return the list of fields without duplicates
+  return Array.from(new Set(fields))
+}
+
+function loadLocalExtensionsIdentifiersBreakdown(
+  localRegistration: IdentifiersExtensions,
+  localSourceToCreate: LocalSource[],
+): ExtensionIdentifiersBreakdown {
+  const identifiersToUpdate = Object.keys(localRegistration)
+  const identifiersToCreate = localSourceToCreate.map((source) => source.localIdentifier)
+  return {
+    onlyRemote: [] as string[],
+    toCreate: [] as string[],
+    toUpdate: [...identifiersToUpdate, ...identifiersToCreate],
+    fromDashboard: [] as string[],
```

Comment:
Shouldn't `identifiersToCreate` be in the `toCreate` array?

Reply:
As decided in the [figma](https://www.figma.com/file/cTu9Ag52V4IdqZe1cij6kK/Versioned-app-config?type=design&node-id=9657-5889&mode=design&t=J4Va6ecVDXCIZhkk-0) 

<img src="https://github.com/Shopify/cli/assets/105213827/829c2239-d0a6-4bdb-a5b4-3c135c081fc4" width=400/>)

when you run the `deploy` command without release, the cli should display the list of local fields